src/magic_folder/test/test_local_snapshot.py

```diff
@@ -289,15 +290,17 @@ def test_create_snapshots(self, filenames, data_strategy):
 
         self.assertThat(self.db.get_all_localsnapshot_paths(), HasLength(len(files)))
         for (file, content) in files:
-            mangled_filename = path2magic(file.asTextMode(encoding="utf-8").path)
+            mangled_filename = mangle_path_segments(
+                file.asTextMode("utf8").segmentsFrom(self.magic.asTextMode("utf8"))
```

Comment:
I guess this may be fodder for a separate PR, but it feels like `.asTextMode` should be something that is happening once, closer to the edge of the system, rather than in the middle of slightly complex logic.

Reply:
I don't really get why we're calling `.asXMode()` where we are as it is .. so happy to have a better way to determine this (I added the `.asTextMode` to self.magic because I presume segmentsFrom needs them both in the same mode).

Reply:
> so happy to have a better way to determine this

My feeling would be:
- whenever we construct a FilePath (or receive one from library code), we use `asTextMode` on it (unless we know we are constructing it from a `unicode`)
- we only use `asBytesMode` before accessing `.path` when passing the result to a function that expects `bytes`.

But, as I said, we don't need to handle that in this PR.